File: Server/EC.Common/DTO/BasketDTO.cs
using EC.Common.Tools;

namespace EC.Common.DTO
{
    public class BasketDTO
    {
        public UnitType Unit { get; set; }
        public DateTime CreateDateEn { get; set; }
        public decimal Price => Quantity * Fee;
        public decimal Fee { get; set; }
        public DateTime StatusDateTimeEn { get; set; }
        public int Quantity { get; set; }
        public string Title { get; set; }
        public string UnitType => Unit.ToUnitText();
        public Guid basketId { get; set; }
        public string CustomerName { get; set; }
        public string EmployeeName { get; set; }
        public decimal TotalPrice { get; set; }
        public string CreateDate => CreateDateEn.ToPersianDateTime();
        public string StatusDateTime => StatusDateTimeEn.ToPersianDateTime();
        public BasketStatus Status { get; set; }
    }

    public class BasketDetailDTO
    {
        public UnitType UnitType { get; set; }
        public string Title { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public string Unit => UnitType.ToUnitText();
        public decimal TotalPrice => UnitPrice * Quantity;
    }
}

File: Server/EC_Server/DTO/TsModels/BasketDTO.ts

module Server.DTO {

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/

    
    export class BasketDTO {
        
        // BASKETID
        public basketId: string = "00000000-0000-0000-0000-000000000000";
        // CUSTOMERNAME
        public customerName: string = null;
        // EMPLOYEENAME
        public employeeName: string = null;
        // TOTALPRICE
        public totalPrice: number = 0;
        // CREATEDATE
        public createDate: string = null;
        // STATUSDATETIME
        public statusDateTime: string = null;
        // STATUS
        public status: BasketStatus = null;
    }
    export class BasketDetailDTO {
        
        // TITLE
        public title: string = null;
        // QUANTITY
        public quantity: number = 0;
        // UNITPRICE
        public unitPrice: number = 0;
        // UNIT
        public unit: string = null;
        // TOTALPRICE
        public totalPrice: number = 0;
    }  
}